doc/mgmt/Management Framework.md

```diff
@@ -0,0 +1,2029 @@
+* [8 Warm Boot Support](#8-warm-boot-support)
+* [9 Scalability](#9-scalability)
+* [10 Unit Test](#10-unit-test)
+
+
+## List of Tables
+
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+## Revision
+
+| Rev |     Date    |       Author            | Change Description                |
+|:---:|:-----------:|:-----------------------:|-----------------------------------|
+| 0.1 | 06/13/2019  | Anand Kumar Subramanian | Initial version                   |
+| 0.2 | 07/05/2019  | Prabhu Sreenivasan      | Added gNMI, CLI content from DELL |
+| 0.3 | 08/05/2019  | Senthil Kumar Ganesan   | Updated gNMI content          |
+| 0.4 | 08/07/2019  | Arun Barboza            | Clarifications on Table CAS   |
+| 0.5 | 08/07/2019  | Anand Kumar Subramanian | Translib Subscribe support    |
+| 0.6 | 08/08/2019  | Kwangsuk Kim            | Updated Developer Workflow and CLI sections |
+| 0.7 | 08/09/2019  | Partha Dutta            | Updated Basic Approach under Design Overview |
+| 0.8 | 08/15/2019  | Anand Kumar Subramanian | Addressed review comments     |
+| 0.9 | 08/19/2019  | Partha Dutta            | Addressed review comments related to CVL    |
+| 0.10 | 09/25/2019  | Kwangsuk Kim           | Updated Transformer section |
+| 0.11 | 09/30/2019  | Partha Dutta           | Updated as per SONiC YANG guideline    |
+
+## About this Manual
+
+This document provides general information about the Management framework feature implementation in SONiC.
+
+## Scope
+
+This document describes the high level design of Management framework feature.
+
+## Definition/Abbreviation
+
+### Table 1: Abbreviations
+
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+| CVL                      | Config Validation Library           |
+| NBI                      | North Bound Interface               |
+| ABNF                     | Augmented Backus-Naur Form          |
+| YANG                     | Yet Another Next Generation         |
+| JSON                     | Java Script Object Notation         |
+| XML                      | eXtensible Markup Language          |
+| gNMI                     | gRPC Network Management Interface   |
+| YGOT                     | YANG Go Tools                       |
+
+## 1 Feature Overview
+
+Management framework is a SONiC application which is responsible for providing various common North Bound Interfaces (NBIs) for the purposes of managing configuration and status on SONiC switches. The application manages coordination of NBI’s to provide a coherent way to validate, apply and show configuration.
+
+### 1.1 Requirements
+
+* Must provide support for:
+
+    1. Standard [YANG](https://tools.ietf.org/html/rfc7950) models (e.g. OpenConfig, IETF, IEEE)
+    2. Custom YANG models ([SONiC YANG](https://github.com/Azure/SONiC/blob/master/doc/mgmt/SONiC_YANG_Model_Guidelines.md))
+    3. Industry-standard CLI
```

Comment:
Can you please define "Industry-standard CLI"?

Reply:
"Industry-standard CLI" refers to mainly CISCO style CLI.

Reply:
Can we put it in the document explicitly?

Reply:
Industry-standard CLI is not a custom construct and it is widely used in the industry.

Reply:
I think it's better to have good defined spec. Do we try to avoid Cisco in this document?

Reply:
Industry standard CLI reference[https://www-01.ibm.com/support/docview.wss?uid=isg3T7000651&aid=1](url)

Reply:
The link you provided describes CLI available on some IBM equipment. And it's name ISCLI, I'm not sure how is this compatible with Cisco.
I think it's better to change wording from "industry standard CLI" to something "Cisco-like CLI". Because there is no such thing like a standard for CLI in the industry.

Reply:
Updated the md file to have cisco like cli